### ISSUE TITLE
Rule proposal: consistent-type-imports / add mixed-type-import prefer

### ISSUE THREAD

**Issue by cscheffauer** [typescript-eslint/typescript-eslint#4506]
### Description

As mentioned and already added as a [PR](https://github.com/typescript-eslint/typescript-eslint/pull/2852), it would be super beneficial to actually have a `mixed-type-import` option or an additional flag on the `consistent-type-imports` rule.

The benefit of having this would be that mixed imports stay as they are, but type-only imports are getting prefixed with the `type` keyword as intended. With the `type-imports` prefer types are getting stripped out of mixed imports and unintentionally increase the lines of code in a file. It should be in the hands of the consumers of this rule to decide whether types are stripped out and new lines are added or not.

Since this is a 100% community-driven project as mentioned in your README.md it would be great if you guys could consider adding that option.

Thanks!

### Fail

```ts
import { A, B } from 'foo';
type T = A;
type U = B;
```

### Pass

```ts
import { A, B } from 'foo';
type T = A;
const b = B;
```

**Comment by bradzacher** [comment]
With [TS's inline type specifier](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names), the lint rule will now accept
```ts
import { type A, B } from 'foo';
type T = A;
const b = B;
```
[play](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=PQKgBApgzgNglgOwC5gAJIJ4AdoGMBOcWSAtNPMsLgPYJRxRITImY4lwC2W1+SUALjAAiCPny9hYEMABQsrjz5gA3mDYQwAMQDCAGjA7q3WsxQBfMADMJnMAHJ8EAIa4k9gNyyNYACpgAXm0dLxo6FAAjQMNjHgQzLyA&rules=N4XyA&tsConfig=N4XyA)

So now you can have the best of both worlds.
We won't be adding any option to the lint rule which allows imports only used as a type to remain as a value.
